sql/spatial.cc

```diff
@@ -3538,6 +3707,50 @@ bool Gis_geometry_collection::get_data_as_json(String *txt, uint max_dec_digits,
 }
 
 
+int Gis_geometry_collection::is_valid(int *valid) const
+{
+  uint32 num_on_geometries;
+  if(this->num_geometries(&num_on_geometries))
+    return 1;
+
+  String *swkb= new String("", 0, &my_charset_bin);
+  uint32 srid= uint4korr(swkb->ptr());
+  Geometry_buffer buff_temp;
+
+  if (swkb->reserve(SRID_SIZE, 512))
+    return 1;
+  swkb->q_append(srid);
+  *valid= 1;
+  for (uint32 i= 1; i <= num_on_geometries; i++)
+  {
+    // extract the i-th geometry
+
+    Geometry *temp;
+    swkb->set_charset(&my_charset_bin);
+    swkb->length(0);
+
+    if(this->geometry_n(i, swkb))
+      return 1;
+
+    temp= Geometry::construct(&buff_temp, swkb->ptr(), swkb->length());
+    if(!temp)
+      return 1;
+
+    int tempValid;
+    if(temp->is_valid(&tempValid))
```

Comment:
style nit: space between `if` and `(condition)` please.

Reply:
Done